LinkedIdeas/CanvasViewController.swift

```diff
@@ -219,15 +305,31 @@ extension CanvasViewController {
   }
   
   override func mouseDragged(with event: NSEvent) {
+    if (dragCount <= 3) {
+      Swift.print("[mouseDragged]")
+      dragCount += 1
+    }
+    
     let point = convertToCanvasCoordinates(point: event.locationInWindow)
     
     // Decision: which actions to trigger
     // given the context and events that happen
     
     switch currentState {
     case .selectedElement(let element):
-      guard let concept = element as? Concept else { return }
-      drag(concept: concept, toPoint: point)
+      if (event.modifierFlags.contains(.shift) || didShiftDragStart) {
+        guard let concept = element as? Concept else { return }
+        creationArrowForLink(toPoint: point)
+        didShiftDragStart = true
+        if let hoveredConcepts = clickedConcepts(atPoint: point) {
+          select(elements: hoveredConcepts)
+        } else {
+          unselect(elements: document.concepts.filter { $0 != concept })
+        }
+      } else {
+        guard let concept = element as? Concept else { return }
```

Comment:
I think you can move these checks out of the `if` statements to only check for it once.